environment-pipelines/tests/unit.tftest.hcl

```diff
@@ -147,6 +147,13 @@ override_data {
   }
 }
 
+override_data {
+  target = data.aws_iam_policy_document.iam
+  values = {
+    json = "{\"Sid\": \"IAM\"}"
+  }
+}
```

Comment:
At present, I think we could accidentally change the policy and nothing would fail.

Should we be asserting on the actual value of the policy?

Reply:
You're right we can change the policy and nothing would fail at present.

There's currently a bug outstanding in TF ([36700](https://github.com/hashicorp/terraform-provider-aws/issues/36700)), as the policy isn't computed when using the mock AWS provider. So we can't properly test the policy has been computed correctly at present

Reply:
I think maybe reference that bug in a comment explaining why it's not tested at present.

Could we test it if we had end to end tests that did an apply?

Reply:
Comment added (next to tests rather than override). We could test is we had end to end tests as the apply would use the policy AWS computed for the test.

Reply:
Approving in a moment.

@antroy-madetech / @JohnStainsby Can/should we add end to end tests for the pipelines?